packages/MainUI/components/Toolbar/SearchPortal.tsx

```diff
@@ -0,0 +1,65 @@
+import React from 'react';
+import { Box, Portal, Theme } from '@mui/material';
+import TextInputAutocomplete from '@workspaceui/componentlibrary/src/components/Input/TextInput/TextInputAutocomplete';
+import { useStyle } from './styles';
+
+interface Position {
+  top: string;
+  right: string;
+}
+
+interface SearchPortalProps {
+  isOpen: boolean;
+  searchValue: string;
+  onSearchChange: (value: string) => void;
+  onClose: () => void;
+  placeholder?: string;
+  position?: Position;
+  width?: string;
+  autoCompleteTexts?: string[];
+  theme?: Theme;
+}
+
+const SearchPortal: React.FC<SearchPortalProps> = ({
+  isOpen,
+  searchValue,
+  onSearchChange,
+  onClose,
+  placeholder,
+  autoCompleteTexts = [],
+}) => {
+  const { styles } = useStyle();
+
+  if (!isOpen) return null;
```

Comment:
place this if ater the `const handleKeyDown` and `const handleBlur` declarations